src/components/Compare/CompareTable.tsx

```diff
@@ -52,9 +53,24 @@ const CompareTable = (props: {
   const locationsWithNull = partitionedLocations[1];
   let sortedLocationsArr = sortedLocationsWithValue.concat(locationsWithNull);
 
+  const populationSort = () => {
+    return sortBy(
+      props.locations,
+      location => location.locationInfo.population,
+    );
+  };
+
+  if (sortByPopulation) {
+    sortedLocationsArr = populationSort();
+  }
+
   if (sortDescending) {
-    sortedLocationsWithValue.reverse();
-    sortedLocationsArr = sortedLocationsWithValue.concat(locationsWithNull);
+    if (sortByPopulation) {
+      sortedLocationsArr.reverse();
+    } else {
+      sortedLocationsWithValue.reverse();
+      sortedLocationsArr = sortedLocationsWithValue.concat(locationsWithNull);
+    }
```

Comment:
Maybe the sorting logic can be unified a bit


```tsx
function sortLocationBy(locations: L[], getValue: (location: L) => number | undefined;) {
   const [locationsWithValue, locationsWithoutValue] = _.partition(locations, getValue);
   const sortedLocationsAsc = _.sortBy(sortedLocation, getValue);
   const sortedLocations = sortDescending ? _.reverse(sortedLocationsAsc) : sortedLocationsAsc;
   return [...sortedLocation, ...locationsWithoutValue];
}

const getPopulation = (location:L) => location?.locationInfo?.population;
const getMetricValue = (location:L) => location.metricsInfo.metrics[sorter].value;

// ... later
if (sortByPopulation) {
  sortedLocationsArr = sortLocationsBy(props.locations, getPopulation, sortDescending);
} else {
  sortedLocationsArr = sortLocationsBy(props.locations, getMetricValue, sortDescending);
}
```

Ignore if this is too much trouble!